pkgs/build-support/kernel/make-initrd-ng/src/main.rs

```diff
@@ -85,12 +83,25 @@ fn copy_file<P: AsRef<Path> + AsRef<OsStr>, S: AsRef<Path>>(
             }
         }
         if !found {
-            // glibc makes it tricky to make this an error because
+            // glibc makes this tricky because
             // none of the files have a useful rpath.
-            println!("Warning: Couldn't satisfy dependency {} for {:?}", line, OsStr::new(&source));
+            if line != "libc.so.6" && !line.starts_with("ld-linux-") {
+                return Err(Error::new(ErrorKind::Other, format!("failed: couldn't satisfy dependency {} for {:?}", line, OsStr::new(&source))))
+            }
         }
     }
 
+    // Make file writable to strip it
+    let mut permissions = fs::metadata(&target)?.permissions();
+    permissions.set_readonly(false);
+    fs::set_permissions(&target, permissions)?;
+
+    // Strip further than normal
+    if !Command::new("strip").arg("--strip-all").arg(OsStr::new(&target)).output()?.status.success() {
```

Comment:
What happens when `strip` isn't on `PATH`? This matters for the `shutdownRamfs` stuff, which currently uses `systemd.services.generate-shutdown-ramfs.path` to provide `ldd` and `patchelf`. Maybe we should make it a wrapper?